examples/simulatedannealing.rs

```diff
@@ -64,27 +64,23 @@ impl ArgminOp for Rosenbrock {
     /// This function is called by the annealing function
     fn modify(&self, param: &Vec<f64>, temp: f64) -> Result<Vec<f64>, Error> {
         let mut param_n = param.clone();
+        let mut rng = self.rng.lock().unwrap();
+        let distr = Uniform::from(0..param.len());
         // Perform modifications to a degree proportional to the current temperature `temp`.
         for _ in 0..(temp.floor() as u64 + 1) {
             // Compute random index of the parameter vector using the supplied random number
             // generator.
-            let mut rng = self.rng.lock().unwrap();
-            let idx = (*rng).gen_range(0..param.len());
+            let idx = rng.sample(distr);
 
             // Compute random number in [0.1, 0.1].
-            let val = 0.1 * (*rng).gen_range(-1.0..1.0);
+            let val = rng.sample(Uniform::new_inclusive(-0.1, 0.1));
 
             // modify previous parameter value at random position `idx` by `val`
-            let tmp = param[idx] + val;
+            param_n[idx] += val;
 
             // check if bounds are violated. If yes, project onto bound.
-            if tmp > self.upper_bound[idx] {
-                param_n[idx] = self.upper_bound[idx];
-            } else if tmp < self.lower_bound[idx] {
-                param_n[idx] = self.lower_bound[idx];
-            } else {
-                param_n[idx] = param[idx] + val;
-            }
+            param_n[idx] = param_n[idx].min(self.upper_bound[idx]);
```

Comment:
Just realized we could use the newish [`f64::clamp`](https://doc.rust-lang.org/std/primitive.f64.html#method.clamp) function instead

Reply:
I would also prefer `clamp` as it seems to be the most clear way of achieving this. Thanks for mentioning it, I didn't know about this method.